File: src/main/java/io/github/kingg22/api_vacunas_panama/service/IVacunaService.java
package io.github.kingg22.api_vacunas_panama.service;

import io.github.kingg22.api_vacunas_panama.response.IApiContentResponse;
import io.github.kingg22.api_vacunas_panama.web.dto.DosisDto;
import io.github.kingg22.api_vacunas_panama.web.dto.InsertDosisDto;
import io.github.kingg22.api_vacunas_panama.web.dto.VacunaFabricanteDto;

import java.util.List;
import java.util.UUID;

public interface IVacunaService {
    IApiContentResponse createDosis(InsertDosisDto insertDosisDto);

    List<VacunaFabricanteDto> getVacunasFabricante();

    List<DosisDto> getDosisByIdPacienteIdVacuna(UUID idPaciente, UUID idVacuna);
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/modules/vacuna/service/IVacunaService.kt
package io.github.kingg22.api.vacunas.panama.modules.vacuna.service

import io.github.kingg22.api.vacunas.panama.modules.vacuna.dto.DosisDto
import io.github.kingg22.api.vacunas.panama.modules.vacuna.dto.InsertDosisDto
import io.github.kingg22.api.vacunas.panama.modules.vacuna.dto.VacunaFabricanteDto
import io.github.kingg22.api.vacunas.panama.response.ApiContentResponse
import java.util.UUID

interface IVacunaService {
    fun createDosis(insertDosisDto: InsertDosisDto): ApiContentResponse

    val vacunasFabricante: List<VacunaFabricanteDto>

    fun getDosisByIdPacienteIdVacuna(idPaciente: UUID, idVacuna: UUID): List<DosisDto>
}
